source/source/nebula/ecs/system/DrawSpriteSystem.hx

```diff
@@ -26,10 +26,19 @@ class DrawSpriteSystem extends AbstractSystem
             var sprite:SpriteComponent = entity.get(SpriteComponent);            
             if (sprite.sprite == null)
             {
-                var s:FlxSprite = new FlxSprite();
-                s.loadGraphic(sprite.image);
-                sprite.sprite =  s;
-                this.state.add(s);
+                if(sprite.isRepeating)
+                {
+                    var s:FlxBackdrop = new FlxBackdrop(sprite.image);
+                    sprite.sprite =  s;
+                    this.state.add(s);
```

Comment:
Could you refactor these two lines so that there's no duplication? Maybe something like:

```
sprite.sprite = new FlxBackdrop(sprite.image);
{ else ... }
this.state.add(s);